### PR TITLE
 Move to the python:alpine docker image 

### LINKED ISSUE #498: Move to the python:alpine docker image
The default image we are currently using is super fat (around 700MB) as it is based on a full fletched debian. There's a lot of movement to images based on alpine going on and the official image now supports that too.

Alpine uses musl libc instead of glibc, so that might be a problem for packages that have tight dependencies with libc.

### PR DESCRIPTION
**Pull request by browniebroke**
Fixes #498 

- Switch the base images for local and production to alpine
- Install extra dependencies for psycopg2, Pillow and cffi
- Change shebang for shell scripts to use sh instead of bash, as bash isn't installed
- Remove `function` keyword from shell scripts, it is unsupported in sh
- Migrate `django` group and user creation to the Alpine syntax

I've started from a generated project, which I've migrated by running `docker-compose -f local.yml build` and `docker-compose -f production.yml build`. The tests are passing with the local config and I can start with the production config.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/compose/local/django/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,16 @@
-FROM python:3.5
+FROM python:3.5-alpine
```
Comment (webyneter) [inline {{cookiecutter.project_slug}}/compose/local/django/Dockerfile]: Make it `FROM python:3.6-alpine`.

File: {{cookiecutter.project_slug}}/compose/production/django/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,18 @@
-FROM python:3.5
+FROM python:3.5-alpine
```
Comment (webyneter) [inline {{cookiecutter.project_slug}}/compose/production/django/Dockerfile]: Make it `FROM python:3.6-alpine`.

**Comment by jayfk** [comment]
Another thing to keep in mind is support for translations.

**Comment by browniebroke** [comment]
Thanks for pointing this out, I'll further testing around this.

**Comment by browniebroke** [comment]
I think it's still missing something to work out of the box, a `locale` folder. Suggest to add it with a README in it?

**Comment by browniebroke** [comment]
I think it's merge-ready from my end. Anything to add?

File: tests/test_docker.sh [hunk 3]
```diff
@@ -19,3 +19,6 @@ docker-compose -f local.yml run django python manage.py test
 
 # return non-zero status code if there are migrations that have not been created
 docker-compose -f local.yml run django python manage.py makemigrations --dry-run --check || { echo "ERROR: there were changes in the models, but migration listed above have not been created and are not saved in version control"; exit 1; }
+
+# Test support for translations
+docker-compose -f local.yml run django python manage.py makemessages
```
Comment (webyneter) [inline tests/test_docker.sh]: I'd make it `docker-compose -f local.yml run --rm django python manage.py makemessages`.

**Comment by pydanny** [comment]
is this good to go? I want to make #1484 happen!

**Comment by jayfk** [comment]
This one should be good. 

It isn‘t blocking anything, though. We are already using Python 3.6 in Docker.

**Comment by browniebroke** [comment]
@jayfk is correct, #1418 unblocked it. Going to update this branch from master now.

**Pull request merged**